python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -7301,22 +7299,36 @@ def test_series_keys(ps):
 )
 @pytest.mark.parametrize("sort", [False, True])
 @pytest.mark.parametrize("ignore_index", [True, False])
-def test_dataframe_append_dataframe(df, other, sort, ignore_index):
+def test_dataframe_concat_dataframe(df, other, sort, ignore_index):
     pdf = df
     other_pd = other
 
     gdf = cudf.from_pandas(df)
     other_gd = cudf.from_pandas(other)
 
-    with pytest.warns(FutureWarning, match="append method is deprecated"):
-        expected = pdf.append(other_pd, sort=sort, ignore_index=ignore_index)
-    with pytest.warns(FutureWarning, match="append method is deprecated"):
-        actual = gdf.append(other_gd, sort=sort, ignore_index=ignore_index)
+    expected = pd.concat([pdf, other_pd], sort=sort, ignore_index=ignore_index)
+    actual = cudf.concat([gdf, other_gd], sort=sort, ignore_index=ignore_index)
+
+    # In some cases, Pandas creates an empty Index([], dtype="object") for
```

Comment:
Could you show a example where this happens? I think pandas would also like to return an empty RangeIndex too if the object is empty-like

Reply:
Apologies for an incorrect comment previously. So the actual issue is with how `cudf` & `pandas` represent empty columns. In case of pandas it's always an empty `RangeIndex`, but in cudf it's always an empty `Index`:
```python
In [1]: import cudf

In [2]: import pandas as pd

In [3]: df = cudf.DataFrame()

In [4]: pdf = pd.DataFrame()

In [5]: df.columns
Out[5]: RangeIndex(start=0, stop=0, step=1)

In [6]: pdf.columns
Out[6]: Index([], dtype='object')
```

Due to this difference, when we concat similar dataframes, pandas still returns a `RangeIndex`, whereas cudf will return an empty `Index`. Which is why we needed this special handling here for pytest.

Reply:
Okay cool thanks for the clarification!